packages/server/src/script/fetchMGErrors.ts

```diff
@@ -0,0 +1,105 @@
+severity=(WARNING OR ERROR OR CRITICAL OR ALERT OR EMERGENCY)
+
+timestamp >= "${fromIsoStamp}"
+${untilIsoStamp ? `timestamp <= ${untilIsoStamp}` : ''}`
+
+  console.log('Querying all MG errors in our logs during the period, this might take a while...\n')
+  // The response comes in a nested array, flattening it to ease iterations
+  const entries = (await logging.log('mailgun-log').getEntries({ filter })).flat() as EntryResponse[]
+
+  const resendIds = entries.map(({ data }) => {
+    // Do not resend since the email doesn't exist, apparently the message
+    // for this event vary depending on the destination
+    //
+    // We don't use the delivery-status code here since email providers'
+    // implementation differ slightly from each other, for example at Google
+    // Gmail we have 550 for non existent emails, but on Yahoo the same error
+    // is represented by 554.
+    const message = data['event-data']?.['delivery-status']?.message
+
+    // Message can only be undefined/null in staging/dev environments, this
+    // happens when we've used MG's webhook testing hooks before (they don't
+    // contain delivery-statuses)
+    if (!message) return null
+
+    // Found on production logs
+    if (message.indexOf('No SMTP service') >= 0) return null
+    // At https://documentation.mailgun.com/en/latest/api-events.html?highlight=user%20variables#event-structure
+    // It seems that this is also the default message from Google's Gmail
+    if (message.indexOf('5.1.1 The email account that you tried to reach does not exist.') >= 0) return null
+    // Used by Yahoo
+    if (message.indexOf('delivery error: dd Not a valid recipient') >= 0) return null
+
+    // If the event.reason is on the ignoreReason list we don't resend the email as well
+    if (ignoreEventReason.indexOf(data['event-data'].reason) >= 0) return null
+
+    const metadata: {uid: string} = JSON.parse(data['event-data']['user-variables'].metadata)
+    return metadata.uid
+  }).filter(d => !!d) as string[] // Removes null cases
+
+  return resendIds
+}
+
+/**
+ * @param from The starting point of the query, remember that month 0 == January
+ * @param until The ending point of the query, can be null to query until the current time
+ * @param ignoreEventReason Ignores the reasons inside this array, for example
+ * to avoid fetching the IDs for bounce events just add 'bounce' to this array.
+ */
+const main = async (from: Date, until: Date | null, ignoreEventReason: MailgunEventReason[] = []) => {
+  const now = new Date()
+  const executionTimeStamp = now.valueOf()
+
+  const ids = await fetchMGErrors(
+    from.toISOString(),
+    until?.toISOString() ?? null,
+    ignoreEventReason,
+  )
+
+  const file = `${__dirname}/data/fetchMGErrors-${executionTimeStamp}.json`
+  const fileJSON = {
+    executionTime: now.toISOString(),
+    from: from.toISOString(),
+    until: until?.toISOString() ?? null,
+    ignoreEventReason,
+    ids,
```

Comment:
Like what I was trying to say in the chat (though I think I may have been unclear, sorry :P), can we try to create "categories" of reasons for failures and try to bucket these in each category? We could also have an 'Other' category for something not mapped.  That way I can kind of decide for which 'reasons' it's worth resending the emails.